apis/apps/pub/inplace_update.go

```diff
@@ -101,6 +107,16 @@ type InPlaceUpdateStrategy struct {
 	GracePeriodSeconds int32 `json:"gracePeriodSeconds,omitempty"`
 }
 
+// InPlaceUpdateContainerRestartCount record the restart count of containers
+type InPlaceUpdateContainerRestartCount struct {
+	// Revision is the updated revision hash.
+	Revision string `json:"revision,omitempty"`
+	// RestartCount is the container restart count
+	RestartCount int `json:"restartCount,omitempty"`
```

Comment:
is this field records the count of container at some revision ? if this is the case , cloneset controller can be too busy updating the restart counts,  actually currently the event handler will skip some event if the pod does not change its UpdateReady states. 

```	if coreControl.IsPodUpdateReady(oldPod, cs.Spec.MinReadySeconds) != coreControl.IsPodUpdateReady(curPod, cs.Spec.MinReadySeconds) {
		return false
	}
```
An alternative api design is to record the initial restart count of the container at some revision, cloneset controller only need to update the count during inplace-update, and the user can read the actually counts by decrease the container restart counts by the initial restart Count

Reply:
work on it